client/src/components/Production/ProductionModal.jsx

```diff
@@ -0,0 +1,151 @@
+            <Button onClick={() => setForm('Bike')}>Bike</Button>
+            <div style={formStyle}>
+              <FormControl isRequired style={elementStyle}>
+                <FormLabel>Status</FormLabel>
+                <Select ref={statusRef}>
+                  <Fragment>
+                    <option>Ongoing</option>
+                    <option>Complete</option>
+                    <option>Idle</option>
+                    <option>Cancelled</option>
+                  </Fragment>
+                </Select>
+              </FormControl>
+              <FormControl isRequired style={elementStyle}>
+                <FormLabel>Name</FormLabel>
+                <Input ref={nameRef} />
+              </FormControl>
+              <FormControl isRequired style={elementStyle}>
+                <FormLabel>Description</FormLabel>
+                <Input ref={descriptionRef} />
+              </FormControl>
+            </div>
+            {form === 'Part' ? (
+              <ProductionModalPart ref={partRefs} />
+            ) : (
+              <ProductionModalBike ref={bikeRefs} />
+            )}
+            <div style={formStyle}>
+              <FormControl isRequired style={elementStyle}>
+                <FormLabel>Color</FormLabel>
+                <Input ref={colorRef} />
+              </FormControl>
+              <FormControl isRequired style={elementStyle}>
+                <FormLabel>Quantity</FormLabel>
+                <NumberInput min={1} defaultValue={1}>
+                  <NumberInputField ref={quantityRef} />
+                  <NumberInputStepper>
+                    <NumberIncrementStepper />
+                    <NumberDecrementStepper />
+                  </NumberInputStepper>
+                </NumberInput>
+              </FormControl>
+              <FormControl isRequired style={elementStyle}>
+                <FormLabel>Profit Margin</FormLabel>
+                <NumberInput min={1.1} defaultValue={1.1}>
+                  <NumberInputField ref={profitMarginRef} />
+                  <NumberInputStepper>
+                    <NumberIncrementStepper />
+                    <NumberDecrementStepper />
+                  </NumberInputStepper>
+                </NumberInput>
+              </FormControl>
+            </div>
+            <div style={formStyle}>
+              <FormControl isRequired style={elementStyle}>
+                <FormLabel>Start Date</FormLabel>
+                <TextField onChange={handleStartDateInput} type="date" value={startDate} />
+              </FormControl>
+              <FormControl isRequired style={elementStyle}>
+                <FormLabel>End Date</FormLabel>
+                <TextField onChange={handleEndDateInput} type="date" value={endDate} />
+              </FormControl>
+            </div>
+            <FormControl isRequired style={elementStyle}>
+              <FormLabel>Assembly Machine</FormLabel>
+              <Input ref={assemblyMachineRef} />
```

Comment:
You will link `Assembly Machine` with a machine from the scheduling page in the future right?

Reply:
Yes, next ticket. I can also do it on this PR. Either way.

Reply:
@SiuYe Do it next ticket, I just wanted to confirm

Reply:
Next ticket it is. 👍